src/Controls/src/Core/VisualElementProfiler/VisualElementProfiler.cs

```diff
@@ -0,0 +1,291 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Microsoft.Maui.Controls;
+
+// TODO: NET10 make this public
+internal interface IVisualElementProfiler
+{
+	void Attach(VisualElement element);
+	void Detach(VisualElement element);
+	bool TryGetStats(VisualElement element, [NotNullWhen(true)] out ILayoutPassStats? stats);
```

Comment:
I would like to include two suggestions here:

1. `public Func<VisualElement, bool> ElementFilter { get; set; }` - This would allow filtering based on specific VisualElement criteria, enabling more precise customization.
2. `public EventHandler<PerformanceStatsEventArgs> OnStatsUpdated { get; set; }` - Be able to subscribe and get real time data.